ratelimiter/ratelimiter.go

```diff
@@ -20,33 +20,38 @@ const (
 )
 
 type RatelimiterEntry struct {
-	sync.Mutex
+	mu       sync.Mutex
 	lastTime time.Time
 	tokens   int64
 }
 
 type Ratelimiter struct {
-	sync.RWMutex
-	stopReset chan struct{}
+	mu      sync.RWMutex
+	timeNow func() time.Time
```

Comment:
Does this need to be inside the struct, using memory and indirection for something that's essentially static?

Would build tags be a better solution to have this done compile time?

Reply:
It is possible to use a global variable but it is much more fiddly to make the tests robust under the race detector. I considered it, but there is currently only one Ratelimiter object per Device. 

Given that in practice the number of devices is small, I don't think it's worth the extra lines of code right now. Happy to change in a followup commit.

Reply:
> It is possible to use a global variable but it is much more fiddly to make the tests robust under the race detector.

Why? The test would just set it once in the beginning before usage, right?